File: tests/testStaticMethods.java
public class testStaticMethods{
    public static int getNum(){
        return (4 + 4) * 12;
    }
    public static int getAge(){
        return 5;
    }
    public static int getCount(){
        return 40000;
    }
    public int getNumber(){
        return 6;
    }

    public static void main (String [] args)
    {
        System.out.println(Andrew.getCount());
        Andrew a = new Andrew("lolz");
        System.out.println(a.getCount());
        System.out.println(a.getAge());
        System.out.println(a.getLink());
        System.out.println(getNum());
        System.out.println(getAge());
        System.out.println(getCount());

    }
}

class Andrew{
    String name;
    int age;
    static int count = 4;
    public Andrew(String n){
        this.name = n;
        this.age = count + 1;
    }

    public static int getCount(){
        return count;
    }

    public int getAge(){
        return this.age;
    }

    public int getLink(){
        return getAge() + getCount();
    }
}

File: tests/translated/testStaticMethods.cc
#include "testStaticMethods.h"
#include <iostream>

_testStaticMethods::_testStaticMethods(): __vptr(&__vtable){} 

int _testStaticMethods::getNum(){
return ((4 + 4) * 12);

}

int _testStaticMethods::getAge(){
return 5;

}

int _testStaticMethods::getCount(){
return 40000;

}

int _testStaticMethods::getNumber(testStaticMethods __this){
return 6;

}



void _testStaticMethods::__delete(_testStaticMethods* __this){
delete __this;
}

int main(void) {
std::cout << ({_testStaticMethods::getNum();}) << std::endl;
std::cout << ({_testStaticMethods::getAge();}) << std::endl;
std::cout << ({_testStaticMethods::getCount();}) << std::endl;
return 0;
}

_testStaticMethods_VT _testStaticMethods::__vtable;

Class _testStaticMethods::__class() { 
 static Class k = new java::lang::__Class(__rt::literal("testStaticMethods"), java::lang::__Object::__class());
return k;
}

